packages/m2m-gateway/src/services/tenantService.ts

```diff
@@ -1,8 +1,46 @@
+import { WithLogger } from "pagopa-interop-commons";
+import { m2mGatewayApi } from "pagopa-interop-api-clients";
 import { PagoPAInteropBeClients } from "../clients/clientsProvider.js";
+import { M2MGatewayAppContext } from "../utils/context.js";
+import { toM2MTenant } from "../api/tenantApiConverter.js";
 
 export type TenantService = ReturnType<typeof tenantServiceBuilder>;
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function tenantServiceBuilder(_clients: PagoPAInteropBeClients) {
-  return {};
+export function tenantServiceBuilder({
+  tenantProcessClient,
+}: PagoPAInteropBeClients) {
```

Comment:
Avoid destructuring here, this breaks the tests because destructuring creates a copy of the object, so when you mock a property of the client in the tests, you will not really modify this tenantProcessClient because it's a different object.
You can use `clients.tenantProcessClient` below

```suggestion
export function tenantServiceBuilder(clients: PagoPAInteropBeClients) {
```